pages/map.js

```diff
@@ -55,11 +55,24 @@ export async function getServerSideProps() {
 export default function Map({ data }) {
   let { users, tags } = data;
   const [filteredUsers, setFilteredUsers] = useState([]);
+  const [selectedTags, setSelectedTags] = useState(new Set());
+
   let results = [];
 
+  const updateSelectedTagsFilter = (tagSelected) => {
+    const updatedSelectedTags = new Set(selectedTags);
+
+    if (updatedSelectedTags.has(tagSelected)) updatedSelectedTags.delete(tagSelected)
+    else updatedSelectedTags.add(tagSelected)
```

Comment:
Please use the same coding style as the rest of the project, for example...

```suggestion
    if (updatedSelectedTags.has(tagSelected)) {
       updatedSelectedTags.delete(tagSelected)
     }
    
    updatedSelectedTags.add(tagSelected)
```